sdk/ai/ai-projects/src/agents/files.ts

```diff
@@ -31,6 +31,23 @@ export async function uploadFile(
   return result.body; 
 }
 
+/** Uploads a file for use by other operations. */
+export async function uploadFileAndPoll(
+context: Client, options: UploadFileParameters, sleepInterval: number,
+): Promise<OpenAIFileOutput> {
+  const result = await context.path("/files").post(options);
+  if (!expectedStatuses.includes(result.status)) {
+      throw createRestError(result);
+  }
```

Comment:
Alternatively, you could do `let file = await createFile(context, options)` since this code is redundant to that method